sdccc/src/main/java/com/draeger/medical/sdccc/TestSuite.java

```diff
@@ -525,24 +525,33 @@ protected void defaultConfigure() {
         return createInjector(configurationModule, new TestRunConfig(testRunDir));
     }
 
-    private static void exit(final int exitCode, final Injector injector, final File testRunDir) {
+    private static void exit(
+            final long numberOfTestFailures, final boolean hadError, final Injector injector, final File testRunDir) {
 
+        final TestRunObserver testRunObserver = injector.getInstance(TestRunObserver.class);
         try {
             LOG.info("Stopping SDCcc");
-            printVerdict(exitCode, testRunDir, injector);
+            printVerdict(numberOfTestFailures, testRunDir, injector);
 
             injector.getInstance(MessageStorage.class).close();
         } catch (final RuntimeException | Error e) {
 
             LOG.error("Unchecked exception during cleanup", e);
-            printVerdict(1, testRunDir, injector);
-            System.exit(1);
+            printVerdict(numberOfTestFailures, testRunDir, injector);
+            System.exit(2); // exitCode != 0 to indicate Error
         }
 
-        System.exit(exitCode);
+        if (numberOfTestFailures > 0) {
+            System.exit(1); // exitCode != 0 to indicate Failure
+        }
+        if (hadError || testRunObserver.isInvalid()) {
+            System.exit(2); // exitCode != 0 to indicate Error
+        } else {
+            System.exit(0); // exitCode == 0 to indicate Success
+        }
```

Comment:
This order does not seem correct to me, exit code 2 cases need to be checked before exit code 1, as a run that has validity issues is worse than a valid run with failed tests

Reply:
Changed the order: When Errors and Failures occur in the same TestRun, then the exitCode now indicates Error, not Failure.